### PR TITLE
Fix vote-validation, clustering NaN, and toggle-race bugs

### PR DESCRIPTION
**Pull request by jss367**
- server: range-check Numerical votes in the vote handler (only Agreement
  and Yes/No were validated), so out-of-range or non-numeric values can no
  longer be stored and skew the summary's average/std-dev/spread. Adds
  min_value/max_value to getQuestionForTopic so the bounds are available.
- clustering: guard the silhouette division against 0/0. When a point was
  identical to every member of both its own and the nearest cluster, the
  resulting NaN poisoned the whole score and silently dropped that k from
  selection.
- server: make toggleResponseVote / toggleResponseReaction atomic by running
  the delete-or-insert in a single transaction, so concurrent toggles from
  the same user can't interleave into the wrong on/off state.
- client: add a cancellation guard to the getBrainstormState effect so a late
  ack for a previous discussion can't overwrite the new discussion's state.
- tests: add a regression test for Numerical vote validation.

https://claude.ai/code/session_01Paww4gyzZJKBWJ36T54dcg

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed brainstorm ratings and reactions being overwritten when navigating away
  * Fixed clustering algorithm calculation errors for optimal cluster selection
  * Added validation to ensure numerical votes fall within configured ranges
  * Prevented vote toggles from becoming inconsistent with concurrent requests

* **Tests**
  * Added test coverage for numerical vote validation edge cases

### CODE REVIEW

**Comment by jss367** [comment]
@codex review

File: server.js [hunk 1]
```diff
@@ -880,6 +880,20 @@ io.on('connection', (socket) => {
...
+      // Numerical votes must be a finite number within the question's configured
+      // range. Without this, a crafted socket message could store out-of-range
+      // (or non-numeric) values that silently skew the average / std-dev / spread
+      // shown in the summary — the range is otherwise only enforced at
+      // create/edit time, never when a vote is cast. getQuestionRange supplies
+      // the 0..100 default for legacy questions with null bounds.
+      if (target.type === 'Numerical') {
+        const num = Number(vote);
+        const { minValue, maxValue } = getQuestionRange({ min_value: target.minValue, max_value: target.maxValue });
+        if (!Number.isFinite(num) || num < minValue || num > maxValue) {
```
[the remark this answers is not in the document]
Comment (jss367) [reply server.js]: Fixed in 035d02e. The handler now requires a real scalar numeric value before the range check — a finite `number`, or a non-blank string that parses cleanly. `''`, `[]`, and `null` (which `Number()` coerces to 0) are now rejected even for ranges that include 0, so `addVote` never stores a non-numeric payload. Added regression coverage for the `''` and `[]` cases.

File: server.js [hunk 2]
```diff
@@ -2951,15 +2967,29 @@ async function toggleResponseVote(topic, responseId, userId) {
...
-  if (deleteResult.rowCount === 0) {
-    await pool.query(
-      'INSERT INTO response_votes (response_id, user_id) VALUES ($1, $2) ON CONFLICT DO NOTHING',
+  // Run the delete-or-insert as one transaction on a single connection so two
+  // concurrent toggles from the same user (double-click, two tabs) can't
+  // interleave their DELETE/INSERT across separate pooled connections and land
+  // in the wrong on/off state.
+  const client = await pool.connect();
+  try {
+    await client.query('BEGIN');
```
[the remark this answers is not in the document]
Comment (jss367) [reply server.js:2985]: Fixed in 035d02e. You're right that a bare transaction doesn't serialize the absent-row case under Read Committed. Both `toggleResponseVote` and `toggleResponseReaction` now take a transaction-scoped advisory lock (`pg_advisory_xact_lock(hashtext(...))`) keyed on `(response, user[, reaction])` at the top of the transaction, so the second concurrent toggle blocks until the first commits, then re-reads the now-current state and flips it correctly — closing both the present-row and absent-row races.

**Comment by jss367** [comment]
@codex review
